### PR TITLE
Bug 2143498: Could not load template while creating VM from catalog

### PR DESCRIPTION
**Pull request by avivtur**
Signed-off-by: Aviv Turgeman <aturgema@redhat.com>
## 📝 Description

falling back to `default` namespace if the namespace is not selected (non-admin user can not click create for all projects so the `default` namespace won't apply as he must select a namespace to create)

### After
https://user-images.githubusercontent.com/67270715/203059012-9ebbd78b-fd5d-4c82-b49e-30d65b2a6473.mp4

### CODE REVIEW

File: src/utils/resources/template/hooks/useProcessedTemplate.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { k8sCreate } from '@openshift-console/dynamic-plugin-sdk';
  */
 export const useProcessedTemplate = (
   template: V1Template,
-  namespace: string,
+  namespace = 'default',
```
Comment (upalatucci) [inline src/utils/resources/template/hooks/useProcessedTemplate.ts:14]: WDYT of creating a constant file under `/src/utils` for the `default` namespace? Also for `getTemplateVMWithNamespace`

Comment (avivtur) [reply src/utils/resources/template/hooks/useProcessedTemplate.ts:14]: I saw we do have one under catalog directory, I'll move it

**Comment by orenc1** [comment]
/cherry-pick release-4.12
https://bugzilla.redhat.com/show_bug.cgi?id=2143964

**Pull request merged**